controllers/user.controller.js

```diff
@@ -0,0 +1,49 @@
+const userService = require("../services/user.service");
+const sessionHelper = require("../helpers/sessionHelper");
+
+async function getMessageToSign(req, res) {
+	const { walletAddress } = req.params;
+	const token = Math.random().toString(36).substring(7);
+	req.session.token = token;
+	const message = `${req.session.token}${walletAddress}`;
+	req.session.message =
+		process.env.NODE_ENV == "test" ? { message } : message;
+	res.json({ message: message });
+}
+
+async function registerUser(req, res) {
+	const { walletAddress } = req.body;
+	try {
```

Comment:
You can make the if else blocks into just one function as registerOrCreateUser() which should use two separate functions, one of them should be what if does there, and the other one is the else part. You gonna call them from registerOrCreateUser function. they all gonna be inside services and controller will call it as one function inside try catch